lib/spack/spack/test/spec_syntax.py

```diff
@@ -472,37 +472,17 @@ def _specfile_for(spec_str, filename):
             [Token(TokenType.PROPAGATED_KEY_VALUE_PAIR, value='cflags=="-O3 -g"')],
             'cflags=="-O3 -g"',
         ),
-        # Way too many spaces
+        # Whitespace is allowed in version lists
+        ("@1.2:1.4 , 1.6 ", [Token(TokenType.VERSION, value="@1.2:1.4 , 1.6")], "@1.2:1.4,1.6"),
+        # But not in ranges. `a@1:` and `b` are separate specs, not a single `a@1:b`.
         (
-            "@1.2 : 1.4 , 1.6 ",
-            [Token(TokenType.VERSION, value="@1.2 : 1.4 , 1.6")],
-            "@1.2:1.4,1.6",
-        ),
-        ("@1.2 :   develop", [Token(TokenType.VERSION, value="@1.2 :   develop")], "@1.2:develop"),
-        (
-            "@1.2 :   develop   = foo",
-            [
-                Token(TokenType.VERSION, value="@1.2 :"),
-                Token(TokenType.KEY_VALUE_PAIR, value="develop   = foo"),
-            ],
-            "@1.2: develop=foo",
-        ),
-        (
-            "% intel @ 12.1 : 12.6 + debug",
-            [
-                Token(TokenType.COMPILER_AND_VERSION, value="% intel @ 12.1 : 12.6"),
-                Token(TokenType.BOOL_VARIANT, value="+ debug"),
-            ],
-            "%intel@12.1:12.6+debug",
-        ),
-        (
-            "@ 12.1 : 12.6 + debug - qt_4",
+            "a@1: b",
             [
-                Token(TokenType.VERSION, value="@ 12.1 : 12.6"),
-                Token(TokenType.BOOL_VARIANT, value="+ debug"),
-                Token(TokenType.BOOL_VARIANT, value="- qt_4"),
```

Comment:
Can we keep these two tests and just remove the versions? I would be fine not allowing spaces after a sigil, but until we do that we should probably test variants and compilers.

Reply:
Yeah, agreed